### ISSUE TITLE
Cursor position reset on denite.nvim quit

### ISSUE THREAD

**Issue by wilywampa** [Shougo/denite.nvim#324]
# Problems summary

Cursor position is reset on quit

## Expected

Cursor should be where it was before exiting denite.nvim, e.g. at the most recent line that was jumped to from grep

## Environment Information (Required!)

 * denite version(SHA1): 7a98f9b

 * OS: macOS Sierra

 * Vim/neovim version: NVIM v0.2.1-548-gda99ded25; Build type: Release

 * `:CheckHealth` result(neovim only):

```
health#deoplete#check
========================================================================
## deoplete.nvim
  - SUCCESS: has("nvim") was successful
  - SUCCESS: has("python3") was successful
  - INFO: If you're still having problems, try the following commands:
    $ export NVIM_PYTHON_LOG_FILE=/tmp/log
    $ export NVIM_PYTHON_LOG_LEVEL=DEBUG
    $ nvim
    $ cat /tmp/log_{PID}
    and then create an issue on github

health#nvim#check
========================================================================
## Configuration
  - SUCCESS: no issues found

## Performance
  - SUCCESS: Build type: Release

## Remote Plugins
  - SUCCESS: Up to date

## terminal
  - INFO: key_backspace (kbs) terminfo entry: key_backspace=^H
  - INFO: key_dc (kdch1) terminfo entry: key_dc=\E[3~

## tmux
  - SUCCESS: escape-time: 50ms
  - INFO: $TERM: screen-256color

health#provider#check
========================================================================
## Clipboard (optional)
  - SUCCESS: Clipboard tool found: pbcopy

## Python 2 provider (optional)
  - INFO: `g:python_host_prog` is not set.  Searching for python2 in the environment.
  - SUCCESS: pyenv found: "/usr/local/Cellar/pyenv/1.0.7/libexec/pyenv"
  - INFO: Executable: /usr/local/bin/python2
  - INFO: Python2 version: 2.7.13
  - INFO: python2-neovim version: 0.1.13
  - SUCCESS: Latest python2-neovim is installed: 0.1.13

## Python 3 provider (optional)
  - INFO: `g:python3_host_prog` is not set.  Searching for python3 in the environment.
  - SUCCESS: pyenv found: "/usr/local/Cellar/pyenv/1.0.7/libexec/pyenv"
  - INFO: Executable: /Users/jniehus/Documents/dev/py3.6.1env/bin/python3
  - INFO: Python3 version: 3.6.1
  - INFO: python3-neovim version: 0.1.13
  - SUCCESS: Latest python3-neovim is installed: 0.1.13

## Ruby provider (optional)
  - INFO: Ruby: ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-darwin16]
  - INFO: Host: /usr/local/bin/neovim-ruby-host
  - SUCCESS: Latest "neovim" gem is installed: 0.4.0

```

## Provide a minimal init.vim with less than 50 lines (Required!)

```vim
scriptencoding utf8
set runtimepath+=/Users/jniehus/Downloads/dev/denite.nvim
nnoremap ,a :<C-u>Denite -no-quit grep<CR>
call denite#custom#map('insert', '<C-q>', '<denite:wincmd:p>')
call denite#custom#map('normal', '<C-q>', '<denite:wincmd:p>')
```

## The reproduce ways from neovim starting (Required!)

 1. Start nvim in the denite.nvim directory (so I know what files you'll be searching)
 2. Type `,agit<CR>` to search denite grep for 'git'
 3. Select any result and jump to it - the cursor should be back in the denite window now
 4. Exit the denite buffer by pressing `<C-q>` from normal mode
 5. Return to the denite buffer with `<C-w>j`
 6. Reactivate the denite buffer with `<CR>` - side question: what's the best way to do an autocommand that resumes the denite buffer on WinEnter that handles -buffer-name correctly?
 6. Quit the denite buffer by pressing `<C-c>` - this will reset the cursor position in whatever file had the grep result

Short version:

 1. See #1 above
 2. Type `,agit<CR><CR><C-q><C-w>j<CR><C-c>`

**Comment by Shougo** [comment]
> what's the best way to do an autocommand that resumes the denite buffer on WinEnter that handles -buffer-name correctly?

You can use `normal "\<CR>"`.

**Comment by Shougo** [comment]
This is the feature.
The cursor position is reset by executing `Denite -resume`.

**Comment by wilywampa** [comment]
The desired behavior for a jump list is to return the cursor to its original position, not the jump location?? What's the point of that? If that is indeed the desired behavior, it should also switch back to the original buffer if the jump changed buffers.

**Comment by wilywampa** [comment]
> You can use `normal "\<CR>"`.

Can you give a full example? `autocmd WinEnter * if & filetype == 'denite' | execute "normal '<CR>'" | endif` works but fails because when using a no-quit source, it opens a second denite window sometimes.

**Comment by Shougo** [comment]
I have tested the autocmd, but it does not work.
You should not define the autocmd.
I cannot guarantee the behavior.

**Comment by Shougo** [comment]
I don't fix the behavior because of #324.  (OH, same person)
